controller/engine_controller.go

```diff
@@ -1814,6 +1828,63 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 	return nil
 }
 
+// updateReplicaRebuildFailedCondition updates the rebuild failed condition if replica rebuilding failed
+func updateReplicaRebuildFailedCondition(log *logrus.Entry, ds *datastore.DataStore, replicaName, errMsg string) error {
+	replica, err := ds.GetReplica(replicaName)
+	if err != nil {
+		return err
+	}
+
+	replicaRebuildFailedReason, conditionStatus, err := getReplicaRebuildFailedReason(ds, replica.Spec.NodeID, errMsg)
+	if err != nil {
+		return err
+	}
+
+	replica.Status.Conditions = types.SetCondition(
+		replica.Status.Conditions, longhorn.ReplicaConditionTypeRebuildFailed,
+		conditionStatus, replicaRebuildFailedReason, errMsg)
+
+	_, err = ds.UpdateReplicaStatus(replica)
+
+	return err
+}
+
+func getReplicaRebuildFailedReason(ds *datastore.DataStore, replicaNodeID, errMsg string) (string, longhorn.ConditionStatus, error) {
+	replicaRebuildFailedReason, conditionStatus, isRebuildingFailedByNetwork := getReplicaRebuildFailedReasonFromError(errMsg)
```

Comment:
Change to

```go
func (ec *EngineController) getReplicaRebuildFailedReason(replica *Replica, errMsg string) (reason string, condition longhorn.ConditionStatus, err error) {
...
```

If having multiple returned variables, suggest named variables.

Reply:
Modified and added named variables.
Thank you.